src/modules/tests/tests.reducer.ts

```diff
@@ -43,26 +32,42 @@ const initialState: TestsModel = {
  */
 export function testsReducer(
   state = initialState,
-  action: Action,
-) {
-  if (action.type === LOAD_PERSISTED_TESTS_SUCCESS) {
-    return (<LoadPersistedTestsSuccess>action).tests;
-  }
+  action: testActions.Types | journalActions.JournalActionTypes) {
 
   const slotId = deriveSlotId(state, action);
-  if (!slotId) {
-    return state;
+
+  switch (action.type){
+    case testActions.LOAD_PERSISTED_TESTS_SUCCESS:
+      return (<testActions.LoadPersistedTestsSuccess>action).tests;
+    case testActions.SET_ACTIVITY_CODE:
+      return {
+        ...state,
+        startedTests: {
+          ...state.startedTests,
+          [slotId]: {
+            ...state.startedTests[slotId],
+            activityCode: action.payload,
+          },
+        },
+      };
+    default:
+      return slotId ? createStateObject(state, action, slotId) : state;
+  }
+}
+
+const deriveSlotId = (state: TestsModel, action: Action): string | null => {
+  if (action instanceof journalActions.StartTest) {
+    return `${action.slotId}`;
   }
+  return (state.currentTest && state.currentTest.slotId) ? state.currentTest.slotId : null;
+};
 
+const createStateObject = (state: TestsModel, action: Action, slotId: string) => {
   return {
     ...state,
     startedTests: {
       ...state.startedTests,
       [slotId]: {
-        // Each sub-reducer deals with state scoped to a specific test and has no knowledge of
```

Comment:
Can we keep this comment in? It's useful for people new to the codebase

Reply:
done :)